src/wasmtime/crates/lind-multi-process/src/signal.rs

```diff
@@ -0,0 +1,165 @@
+                    thread_suicide();
+                },
+                SignalDefaultHandler::Ignore => {
+                    continue;
+                },
+                SignalDefaultHandler::Stop => {
+                    // TODO: support STOP signals
+                    eprintln!("Warning: STOP signal received but currently not supported!");
+                    continue;
+                },
+                SignalDefaultHandler::Continue => {
+                    // TODO: support CONTINUE signals
+                    eprintln!("Warning: CONTINUE signal received but currently not supported!");
+                    continue;
+                }
+            }
+        } else if signal_handler == SIG_IGN as u32 { // ignore the signal
+            continue;
+        } else {
+            // we should invoke user's custom signal handler
+
+            // before invoke the function, let's record the signal callstack information in case user performed
+            // any Asyncify-related operation in signal handler
+            caller.as_context_mut().append_signal_asyncify_data(signal_handler as i32, signo);
+            // invoke the 
+            let invoke_res = signal_func.call(caller.as_context_mut(), (signal_handler as i32, signo));
+            // print errors if any when running the signal handler
+            if let Err(err) = invoke_res {
+                let e = wasi_common::maybe_exit_on_error(err);
+                eprintln!("Error: {:?}", e);
+                // if we encountered any error when executing the signal handler, we should terminate the cage
+                rawposix::interface::epoch_kill_all(cageid);
+                thread_suicide();
+            }
+
+            // first let's check if the signal handler returns due to Asyncify Unwind operation
+            if caller.as_context().get_asyncify_state() == AsyncifyState::Unwind {
+                // if it is, then return immediately
+                return 0;
+            } else {
+                // otherwise, the signal handler returns normally
+
+                // restore signal mask
+                restorer(cageid);
+                // clean up the signal callstack information for Asyncify
+                caller.as_context_mut().pop_signal_asyncify_data(signal_handler as i32, signo);
+            }
+        }
+    }
+    
+    0
+}
+
+// raise a trap to the current thread
+// this is paired with catch_traps function in /crates/wasmtime/src/runtime/vm/traphandlers.rs
+// which will catch the trap raised here and perform the clean up
+pub fn thread_suicide() -> ! {
+    // we raise Trap::Interrupt instead of other trap type
+    // because this is the trap type used by wasmtime's built-in epoch
+    // and epoch is the only possible source of this type of trap
+    let err = Trap::Interrupt;
+    unsafe {
+        raise_trap(err.into());
+    }
+    unreachable!();
+}
+
+// maps each signal to its default handler
```

Comment:
probably link to the posix signal defaults here and explain if anything is different than that spec